### ISSUE TITLE
Renovate no longer making PR and warning in the logs 

### ISSUE THREAD

**Issue by berkeleybross** [renovatebot/renovate#10048]
**What Renovate type, platform and version are you using?**

**Renovate type**
Self hosted in Azure DevOps, vmImage: "ubuntu-20.04"

- [ ] Hosted app
- [x] Self hosted

**Renovate platform**

- [x] Azure DevOps (dev.azure.com)
- [ ] Azure DevOps Server (self-hosted)
- [ ] Bitbucket Cloud (bitbucket.org)
- [ ] Bitbucket Server (self-hosted)
- [ ] Gitea (gitea.com, self-hosted)
- [ ] GitHub (github.com, GitHub Enterprise)
- [ ] GitLab (gitlab.com, self-hosted)

**Renovate version (if self hosting)**

I'm self hosting Renovate with version: ...
25.21.13

**Describe the bug**

Renovate doesnt seem to create PR's anymore, and raises a warning in the logs

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

```
WARN: Error updating branch (repository=AireSuite/DependencyUpdater, branch=renovate/renovate-25.x)
       "err": {
         "code": "ERR_INVALID_ARG_TYPE",
         "message": "The \"path\" argument must be of type string. Received undefined",
         "stack": "TypeError [ERR_INVALID_ARG_TYPE]: The \"path\" argument must be of type string. Received undefined\n    at validateString (internal/validators.js:124:11)\n    at Object.join (path.js:1039:7)\n    at Object.join (/home/vsts/work/1/s/node_modules/upath/build/code/upath.js:51:33)\n    at Object.getAdditionalFiles (/home/vsts/work/1/s/node_modules/renovate/dist/manager/npm/post-update/index.js:398:49)\n    at async Object.processBranch (/home/vsts/work/1/s/node_modules/renovate/dist/workers/branch/index.js:216:33)\n    at async Object.writeUpdates (/home/vsts/work/1/s/node_modules/renovate/dist/workers/repository/process/write.js:25:21)\n    at async Object.update (/home/vsts/work/1/s/node_modules/renovate/dist/workers/repository/process/extract-update.js:110:15)\n    at async Object.renovateRepository (/home/vsts/work/1/s/node_modules/renovate/dist/workers/repository/index.js:66:21)\n    at async Object.start (/home/vsts/work/1/s/node_modules/renovate/dist/workers/global/index.js:110:13)\n    at async /home/vsts/work/1/s/node_modules/renovate/dist/renovate.js:33:24"
       }

```

</details>

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [x] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

Looks like it might be fixed by https://github.com/renovatebot/renovate/releases/tag/25.21.15

However, renovate cant update itself anymore. Should we be version locking renovate ourselves or always using "latest"?

**Comment by berkeleybross** [comment]
Confirmed that it was fixed by manually updating renovate to v25.22.0

I would still appreciate advise on "Should we be version locking renovate ourselves or always using "latest"?" - I currently have renovate version locked in package.json/package-lock.json and have renovate automatically updating itself which was working nicely up until this error :P

**Comment by viceice** [comment]
This is already fixed. We are using renovate dry-run mode to check if renovate can successfully self-update. eg here: https://gitlab.com/renovate-bot/renovate-runner/-/blob/master/.gitlab-ci.yml

**Comment by viceice** [comment]
fixed in [25.21.15](https://github.com/renovatebot/renovate/releases/tag/25.21.15) and [25.21.16](https://github.com/renovatebot/renovate/releases/tag/25.21.16)
